standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -1283,6 +1307,23 @@ public Table getTable(String catName, String dbName, String tableName,
         rollbackTransaction();
       }
     }
+
+    // Get the statistics if requested and if the table can have actual data.
+    try {
+      if (tbl != null &&
+          !TableType.VIRTUAL_VIEW.toString().equals(tbl.getTableType()) &&
```

Comment:
If basic stats is false, there won't be any columns with value true in the map so getColumnHavingStats will return nothing and thus no statistics gets added to Table object. See setBasicStatsState(). Do you think that we should still check BASIC_STATS true?

Reply:
If you see earlier in this method for txn stats, first we validate if the stats are valid wrt to current write_id and validWriteIdList. If not we explicitly set it to false in the local tbl object but the column stats still exist in RDBMS. We shouldn't return it if it is not valid. So, we need to check if tbl->parameters(basic_stats)=true before fetching column stats from RDBMS.

Reply:
We are passing  tbl.getParameters() to getColumnsHavingStats(). If tbl->parameters(basic_stats) = false, tbl->parameters(column_stats) will not have any entries (see setBasicStatsState()) and getColumnsHavingStats() will return an empty column list. getTableColumnStatistic() when gets an empty column list returns null. Hence if tbl->parameters(basic_stats) = false, we will not set column statistics in table. Even if tbl->parameters(basic_stats) = true, we will set column stats only for those columns for which column stats is valid since only those columns are returned by getColumnsHavingStats().

What you are suggesting is correct. The code is already behaving that way.